src/app/wallets/reimburse-fee.ts

```diff
@@ -0,0 +1,57 @@
+import { FeeReimbursement } from "@domain/ledger/fee-reimbursement"
+import { LedgerService } from "@services/ledger"
+import { PriceService } from "@services/price"
+
+export const reimburseFee = async ({
+  liabilitiesAccountId,
+  journalId,
+  paymentHash,
+  maxFee,
+  actualFee,
+  logger,
+}: {
+  liabilitiesAccountId: LiabilitiesAccountId
+  journalId: LedgerJournalId
+  paymentHash: PaymentHash
+  maxFee: Satoshis
+  actualFee: Satoshis
+  logger: Logger
+}): Promise<void | ApplicationError> => {
+  const feeReimbursement = FeeReimbursement(maxFee)
+  const feeDifference = feeReimbursement.getReimbursement({
+    actualFee,
+  })
+  if (feeDifference === null) {
+    logger.warn(
+      `Invalid reimbursement fee for ${{
```

Comment:
logger should be use with: `logger.warn({var}, "message with no variable")`. this make searching/parsing messages in the log much easier in production

Reply:
Sure can do, would it be like this?:
```ts
logger.warn(
  {
    maxFee,
    actualFee,
  },
  `Invalid reimbursement fee`,
)
```

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/580/commits/63eb1d1891e6843e386c0ea12fad485083d639d1